Fix failure of test_dynamo_distributed & test_inductor_collectives

Flamefire opened the pull request:
When CUDA is not available `c10d.init_process_group("nccl"...)` will fail with
> RuntimeError: ProcessGroupNCCL is only supported with GPUs, no GPUs found!

Hence add a corresponding skip marker to the classes deriving from DynamoDistributedSingleProcTestCase next to the `requires_nccl` marker.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @mcarilli @ptrblck @leslie-fang-intel @voznesenskym @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Flamefire committed:
Skip tests using DynamoDistributedSingleProcTestCase without CUDA

When CUDA is not available `c10d.init_process_group("nccl"...)` will
fail with
> RuntimeError: ProcessGroupNCCL is only supported with GPUs, no GPUs found!

Hence add a corresponding skip marker to the classes deriving from it
next to the `requires_nccl` marker.

ezyang reviewed (approved):
nice, thank you

ezyang commented:
@pytorchbot merge

malfet commented:
@pytorchbot merge

malfet commented:
@ezyang please approve workflows before asking for a merge to get signal

ezyang commented:
@Flamefire hurry up and accept your pytorch org invite, should fix the approval problem

Flamefire commented:
Uff, conflicted again. Resolved

ezyang commented:
@pytorchbot merge

Flamefire commented:
The submodule update was not intentional. Rebased and removed it.

ezyang commented:
@pytorchbot merge

Flamefire commented:
Looks like pytest doesn't work well with unittest skip markers and still tries to run those tests? Reverted the changes to those files using `pytest.mark.skipif` and fixed the lintrunner issues after getting it to run locally.

ezyang commented:
cool but needs rebase now

ezyang commented:
@Flamefire I sent you an invite to the org which should avoid the need for approvals

ezyang commented:
@pytorchbot merge

Flamefire commented:
> @Flamefire I sent you an invite to the org which should avoid the need for approvals

Canceled? Havent seen it earlier.

Seems I need to check again what went wrong during rebase. Thought I had checked again with the linter... O.o

ezyang commented:
I resent it, they apparently expire. Look at https://github.com/pytorch

Flamefire commented:
> I resent it, they apparently expire. Look at https://github.com/pytorch

Looks like. And I don't even get a push notification for this and by the time I get the EMail it is too late (time zone differences and such...) In the org overview page I can only follow but not join.

Anyway rebased again, linter issues fixed (looked like a merge conflict resolution issue)

ezyang commented:
@pytorchbot merge

ezyang commented:
@Flamefire OK, so I got word back from the OSS team and apparently the 'canceling' of the invites is by design via some sort of infra they have on their side.

So apparently the correct short term resolution is, when you want to make PRs to pytorch, to push your branches to pytorch/pytorch directly (rather than in your fork). I've gone ahead and given you write access to the repo, you're welcome to reopen this PR if necessary.

Flamefire commented:
> So apparently the correct short term resolution is, when you want to make PRs to pytorch, to push your branches to pytorch/pytorch directly (rather than in your fork). I've gone ahead and given you write access to the repo, you're welcome to reopen this PR if necessary.

Seems I got that invite now. But I don't understand: What is the advantage of pushing to this repo instead of my fork (as otherwise usual)?

Identified the cause of the merge test failure: `HAS_CUDA` in inductor_utils is different and actually requires triton in addition. Wasn't obvious, `HAS_CUDA_TRITON` would have been better...   
Fixed that.

huydhn commented:
> Seems I got that invite now. But I don't understand: What is the advantage of pushing to this repo instead of my fork (as otherwise usual)?

Oh, you can use your fork as usual, the point is that you now have write access to PyTorch repo, and doesn't need to get CI run approved every time.  You can create branch on PyTorch repo too if you want.

malfet commented:
IMO it would be much easier to break this PR down into smaller chunks, as test files are modified often, so you'll almost always have some flaky merge conflicts

Flamefire committed:
Add requires_cuda skip marker to common_utils

Flamefire commented:
> IMO it would be much easier to break this PR down into smaller chunks, as test files are modified often, so you'll almost always have some flaky merge conflicts

Makes sense. Reduced this PR to just fixing the test failures

ezyang commented:
@pytorchbot merge
